src/contentnormalizer.js

```diff
@@ -0,0 +1,157 @@
+	 * In other case data are not modified.
+	 *
+	 * Please notice that presence of `isTransformedWithPasteFromOffice` flag in input data prevent transformation.
+	 * This forbid of running the same normalizer twice or running multiple normalizers over the same data.
+	 *
+	 * @param data input data object it should preserve structure defined in
+	 * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation Clipboard#inputTransformation event}.
+	 */
+	transform( data ) {
+		const html = data.dataTransfer && data.dataTransfer.getData( 'text/html' );
+		const dataReadFirstTime = data.isTransformedWithPasteFromOffice === undefined;
+		const hasHtmlData = !!html;
+
+		if ( hasHtmlData && dataReadFirstTime && this._activationTrigger( html ) ) {
+			this._applyFilters( data );
+			data.isTransformedWithPasteFromOffice = true;
+		}
+	}
+
+	/**
+	 * Adds filter function to Content Normalizer.
+	 * Function is called with configuration object where `data` key keeps reference to input data obtained from
+	 * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation Clipboard#inputTransformation event}
+	 *
+	 * See also: {@link module:paste-from-office/contentnormalizer~FilterFunction}
+	 *
+	 * @param {module:paste-from-office/contentnormalizer~FilterFunction} filterFn
+	 */
+	addFilter( filterFn ) {
+		this._filters.add( filterFn );
+	}
+
+	/**
+	 * Applies filters stored in {@link #_filters} to currently processed data.
+	 *
+	 * @private
+	 * @param {Object} data input data object it should preserve structure defined in
+	 * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation Clipboard#inputTransformation event}.
+	 */
+	_applyFilters( data ) {
+		const writer = new UpcastWriter();
+		const documentFragment = data.content;
+
+		for ( const filter of this._filters ) {
+			filter( { data, documentFragment, writer } );
+		}
+	}
+}
+
+/**
+ * Filter function which is used to transform data of
+ * {@link module:clipboard/clipboard~Clipboard#event:inputTransformation Clipboard#inputTransformation event}.
+ *
+ * Filters are used by {@link module:paste-from-office/contentnormalizer~ContentNormalizer}.
+ *
+ * Examples:
+ *
+ * 	function removeBoldTagWrapper( { documentFragment, writer } ) {
+ * 		for ( const childWithWrapper of documentFragment.getChildren() ) {
+ * 			if ( childWithWrapper.is( 'b' ) && childWithWrapper.getStyle( 'font-weight' ) === 'normal' ) {
+ * 				const childIndex = documentFragment.getChildIndex( childWithWrapper );
+ * 				const removedElement = writer.remove( childWithWrapper )[ 0 ];
+ *
+ * 				writer.insertChild( childIndex, removedElement.getChildren(), documentFragment );
+ * 			}
+ * 		}
+ * 	}
+ *
+ * 	function transformWordContent( { data } ) {
+ * 		const html = data.dataTransfer.getData( 'text/html' );
+ *
+ * 		data.content = _normalizeWordInput( html, data.dataTransfer );
+ * 	}
+ *
+ * @callback module:paste-from-office/contentnormalizer~FilterFunction
```

Comment:
This definition is OK - might be a good starting point for refactoring the Word filters later on. I'll check the docs later on as we need to rewrite the moralizers.